chains/solana/contracts/programs/ccip-router/src/instructions/v1/token_admin_registry.rs

```diff
@@ -5,74 +5,120 @@ use solana_program::{address_lookup_table::state::AddressLookupTable, log::sol_l
 use super::pools::token_admin_registry_writable;
 
 use crate::{
-    AcceptAdminRoleTokenAdminRegistry, AdministratorRegistered, AdministratorTransferRequested,
-    AdministratorTransferred, CcipRouterError, ModifyTokenAdminRegistry, PoolSet,
-    RegisterTokenAdminRegistryViaGetCCIPAdmin, RegisterTokenAdminRegistryViaOwner,
-    SetPoolTokenAdminRegistry, CCIP_TOKENPOOL_CONFIG, CCIP_TOKENPOOL_SIGNER,
-    FEE_BILLING_TOKEN_CONFIG, TOKEN_ADMIN_REGISTRY_SEED,
+    AcceptAdminRoleTokenAdminRegistry, AdministratorTransferRequested, AdministratorTransferred,
+    CcipRouterError, ModifyTokenAdminRegistry, PoolSet, RegisterTokenAdminRegistryByCCIPAdmin,
+    RegisterTokenAdminRegistryByOwner, SetPoolTokenAdminRegistry, CCIP_TOKENPOOL_CONFIG,
+    CCIP_TOKENPOOL_SIGNER, FEE_BILLING_TOKEN_CONFIG, TOKEN_ADMIN_REGISTRY_SEED,
 };
 
 const MINIMUM_TOKEN_POOL_ACCOUNTS: usize = 9;
 
-pub fn register_token_admin_registry_via_get_ccip_admin(
-    ctx: Context<RegisterTokenAdminRegistryViaGetCCIPAdmin>,
-    mint: Pubkey, // should we validate that this is a real token program?
+// Proposes an administrator for the given token as pending administrator
+pub fn ccip_admin_propose_administrator(
+    ctx: Context<RegisterTokenAdminRegistryByCCIPAdmin>,
     token_admin_registry_admin: Pubkey,
 ) -> Result<()> {
+    let token_mint = ctx.accounts.mint.key().to_owned();
     let token_admin_registry = &mut ctx.accounts.token_admin_registry;
+
+    set_pending_administrator(token_mint, token_admin_registry, token_admin_registry_admin)?;
+
+    Ok(())
+}
+
+// Proposes an administrator for the given token as pending administrator
+pub fn owner_propose_administrator(
+    ctx: Context<RegisterTokenAdminRegistryByOwner>,
+    token_admin_registry_admin: Pubkey,
+) -> Result<()> {
+    let token_mint = ctx.accounts.mint.key().to_owned();
+    let token_admin_registry = &mut ctx.accounts.token_admin_registry;
+
+    set_pending_administrator(token_mint, token_admin_registry, token_admin_registry_admin)?;
+
+    Ok(())
+}
+
+fn set_pending_administrator(
+    token_mint: Pubkey,
+    token_admin_registry: &mut Account<'_, crate::TokenAdminRegistry>,
+    new_admin: Pubkey,
+) -> Result<()> {
+    require_neq!(
+        new_admin,
+        Pubkey::new_from_array([0; 32]),
+        CcipRouterError::InvalidTokenAdminRegistryInputsZeroAddress
+    );
+
+    // TODO: Revert if administrator != 0 AlreadyRegistered
```

Comment:
I feel like this TODO should be addressed before merging (should only be a `require` with a custom error, right?) because otherwise the method signature is confusing: it claims to only set the pending administrator but it's also deleting the current one.

Reply:
Side note: Maybe this could be named/commented to clarify this is meant to be used for initialization only: otherwise it might be tempting to erroneously call this from the `transfer` function. Maybe calling this "initial_set_pending_administrator" would help.

Reply:
Oh, yes sorry, this will be handled before merging. Just wanted a quick review to save time. 
I'm going to create a different message for pending admin override, so that check will be there and this internal function can be renamed to reflect the initialization as well. Thanks